dockstore/dockstore-cli
# Move nightly build to CircleCI

**Description**
Moves nightly build against dockstore/dockstore develop snapshot to CircleCI from GitHub actions.

The bug noted that the nightly build against develop was only catching compilation errors. This turned out to be because the GitHub action was only running the unit tests. It was not running the integration tests.

To add the integration tests, it was decided to move the build to CircleCI. This avoids duplication, e.g., decryption for confidential tests, invoking the different Maven profiles, etc.

The trigger to fire this nightly is [here](https://app.circleci.com/settings/project/github/dockstore/dockstore-cli/triggers?return-to=https%3A%2F%2Fapp.circleci.com%2Fprojects%2Fproject-dashboard%2Fgithub%2Fdockstore%2F) -- will finalize it once this is merged.

**Review Instructions**
1. Verify that a build is run nightly against the lastest snapshot of dockstore/dockstore
2. Verify that "regular" builds are building against the version in the pom.xml

To do so, you need to look at the build logs, and look for dockstore-common. 

The build against develop webservice message will look like this, note the `SNAPSHOT`:

```
Downloading from artifacts.oicr.on.ca.ca-snapshots: https://artifacts.oicr.on.ca/artifactory/collab-snapshot/io/dockstore/dockstore-common/1.15.0-SNAPSHOT/dockstore-common-1.15.0-20231201.202101-134.jar
```

The build against the web service version defined in the pom will look like this:

```
Downloading from artifacts.oicr.on.ca.ca-snapshots: https://artifacts.oicr.on.ca/artifactory/collab-snapshot/io/dockstore/dockstore-common/1.15.0-alpha.11/dockstore-common-1.15.0-alpha.11.pom
```

**Issue**
SEAB-5853

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `./mvnw clean install`
